genie-web/src/main/java/com/netflix/genie/web/configs/grpc/GenieGRpcServerAutoConfiguration.java

```diff
@@ -0,0 +1,110 @@
+/*
+ *
+ *  Copyright 2019 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.configs.grpc;
+
+import com.netflix.genie.web.properties.GRpcServerProperties;
+import com.netflix.genie.web.rpc.grpc.servers.GRpcServerManager;
+import io.grpc.BindableService;
+import io.grpc.Server;
+import io.grpc.ServerInterceptor;
+import io.grpc.ServerInterceptors;
+import io.grpc.netty.NettyServerBuilder;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import javax.annotation.PostConstruct;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Controls whether a gRPC server is configured and started for this Genie node or not.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+@Configuration
+@ConditionalOnProperty(value = GRpcServerProperties.ENABLED_PROPERTY, havingValue = "true")
+@EnableConfigurationProperties(
+    {
+        GRpcServerProperties.class,
+    }
+)
+@AutoConfigureAfter(
+    name = {
+        "com.netflix.springboot.grpc.server.GrpcServerAutoConfiguration"
+    }
+)
+@Slf4j
+public class GenieGRpcServerAutoConfiguration {
+
+    private static final int RANDOM_PORT = 0;
+
+    /**
+     * Log that gRPC server is enabled.
+     */
+    @PostConstruct
+    public void postConstruct() {
+        log.info("gRPC server configuration is ENABLED");
+    }
+
+    /**
+     * Create a {@link Server} if one isn't already present in the context.
+     *
+     * @param port               The port this server should listen on
+     * @param services           The gRPC services this server should serve
+     * @param serverInterceptors The {@link ServerInterceptor} implementations that should be applied to all services
+     * @return A Netty server instance based on the provided information
+     */
+    @Bean
+    @ConditionalOnMissingBean(Server.class)
+    public Server gRpcServer(
+        @Value("${grpc.server.port:0}") final int port,
```

Comment:
We don't use `@Value` in the rest of the codebase (that i'm aware of). Any particular reason for straying from convention here?

Reply:
yeah. We've tended to avoid it. I used it here (for now) because:
- We already have a `GRpcServerProperties` class but its prefix is `genie.grpc.server` and I couldn't decide if I wanted to move `port` under there or not and since I couldn't decide I didn't want to either create a new class or rename the existing one
- This is the property set from the grpc library we were using so theoretically I wouldn't have to change internal configuration to plot this in
- It might also be the property of the internal gRPC starters. I have to double check that

Effectively I'm not sure what we'll want to do here and since we weren't using the gRPC port anywhere else (yet) I was comfortable putting it in temporarily while fleshing out the rest of this

I'll add a `TODO` so it's clear I don't intend it to stick around